#164859749 User reset password via email

jmusila opened the pull request:
## What does this PR do?
- Creates user reset password endpoint

## Description of Task to be completed?
- [x] Create user reset password endpoint

## How should this be manually tested?
- Clone the repo to your local machine from here

- cd into the ah-django directory and checkout to the branch `ft-reset-password-viaemail-164859749`

- `cd ah-django`

- git checkout `ft-reset-password-viaemail-164859749`

- Create a virtual environment and activate it:

- `python3 -m virtualenv venv && source venv/bin/activate`

- If you need to install virtualenv first:

- `pip install virtualenv`

- Install the project dependencies

- `pip install -r requirements.txt`

- Make database migrations and start the django server

- `python3 manage.py makemigrations`

- `python3 manage.py migrate`

- `python3 manage.py runserver`

- signup as a new user using a valid email

- try changing passing using the signed up email as the input

- check your inbox to find the send link and click on it to reset the password

- for `receiving a reset link` provide the data to postman in this format:
  ```  
     {
	 "email": "test.user@gmail.com"
     }
  ```
- for `resetting the password` provide the data to postman in this format:
  ```
     {
       "password": "mYp@55ord"
       "password": "mYp@55ord"
     }
  ```
- request type `POST`
## Any background context you want to provide?
- Initially, the application had no reset password endpoint

## What are the relevant pivotal tracker stories?
#164859749

## Screenshots:
![Insomnia – http:--127 0 0 1:8000-api-v1-users-forgot_password_161](https://user-images.githubusercontent.com/42172425/55416740-0c299900-5578-11e9-9ac6-93587f22487d.png)
![Selection_160](https://user-images.githubusercontent.com/42172425/55416770-18adf180-5578-11e9-811b-06180b764235.png)
![Insomnia – http:--127 0 0 1:8000-api-v1-users-forgot_password_166](https://user-images.githubusercontent.com/42172425/55468495-2b720600-560c-11e9-9cf6-22032540fc84.png)
![Selection_164](https://user-images.githubusercontent.com/42172425/55468513-388ef500-560c-11e9-89e6-23ded3804275.png)

philophilo reviewed (changes requested):
All these changes are not necessary, we are dealing with backend alone.

philophilo commented on authors/apps/authentication/urls.py:
```diff
@@ -1,12 +1,17 @@
 
+
 urlpatterns = [
-    url(r'^user/?$', UserRetrieveUpdateAPIView.as_view()),
```
Any reason we are getting rid of this endpoint

jmusila replied:
Yeah I think I will remove that

philophilo replied:
@jonathanmusila the reason

jmusila replied:
I think I got you now, no reason to get rid of it, I think I removed that by mistake, will correct that

philophilo commented on authors/apps/authentication/tests/base_test.py:
```diff
@@ -33,10 +34,38 @@ def setUp(self):
+
+        self.reset_password = {
+            "email": "",
+        }
```
use descriptive names

jmusila replied:
Done

philophilo commented on authors/apps/authentication/tests/base_test.py:
```diff
@@ -33,10 +34,38 @@ def setUp(self):
+
+        self.reset_password1 = {
+            "email": "cow@mammals.milk",
+        }
```
here too

philophilo commented on authors/apps/authentication/tests/test_auser_account.py:
```diff
@@ -11,20 +11,6 @@ class TestUser(BaseTestCase):
 
-    def test_new_user_can_register(self):
-        """
-            A user should be informed of a verification email sent to
```
why do we need to do away with this test

jmusila replied:
The test is still there I just transferred it to file `test_amail_send` I thought my test was depending on it

philophilo replied:
Great...

jmusila committed:
164859749-feat(reset-password-via-email):User reset password via email

- A user is able to receive a reset password link to email
- Using the received link a user can reset the password
- The user receives a success email

[Delivers #164859749]: ft-reset-password-viaemail-164859749
